packages/core/admin/admin/src/content-manager/services/relations.ts

```diff
@@ -49,34 +49,45 @@ const relationsApi = contentManagerApi.injectEndpoints({
         };
       },
       merge: (currentCache, newItems) => {
-        if (currentCache.results && newItems.results) {
-          /**
-           * Relations will always have unique IDs, so we can therefore assume
-           * that we only need to push the new items to the cache.
-           */
-          const existingIds = currentCache.results.map((item) => item.documentId);
-          const uniqueNewItems = newItems.results.filter(
-            (item) => !existingIds.includes(item.documentId)
-          );
-          currentCache.results.push(...uniqueNewItems);
-          currentCache.pagination = newItems.pagination;
+        if (currentCache.pagination && newItems.pagination) {
+          if (currentCache.pagination.page < newItems.pagination.page) {
+            /**
+             * Relations will always have unique IDs, so we can therefore assume
```

Comment:
Is this always true? I'm thinking of situations like https://github.com/strapi/strapi/pull/19681#discussion_r1512943884

Reply:
You still return the numerical `id` right which would be unique regardless of locale & status? assuming they're not polymorphic, which we dont support in the CM.